### ISSUE TITLE
do not save images within a 'display:none' elements

### ISSUE THREAD

**Issue by rychphd** [gildas-lormeau/SingleFile#146]
I have a Tampermonkey script that redefines for some elements
`{ display:none !important; }` and indeed they along with images within are not shown. However when I try to save it either using ScrapBook X with images, or SingleFile -- the images are saved. In case of SingleFile the unwanted images are saved within the html file making it quite large.

So, the problem is that the images that are not displayed as a result of a user script, are still saved by SingleFile.

**Comment by gildas-lormeau** [comment]
@rychphd I just published the version 1.9.52 which should improve things. Does it fix your issue?

**Comment by rychphd** [comment]
Yes! Thank you! No more big blocks of encoded binary data in the .html file and it's much smaller.

((
By the way, I simply don't know how to actually  _remove_ the elements, so I just do 



// ==UserScript==
// ...
// @run-at document-start
// ==/UserScript==
var style = document.createElement('style'); style.innerHTML = \`
    #sidebar { display:none !important; }
    .topbar { display:none !important; }
\`; document.head.appendChild(style);



But it must be possible, because when I enter similar rules into uBlock Origin, SingleFile successfully skipped images therein and saved a small html. Anyways, now that you released a new version it works for my current script and I don't have to port it into uBlock filters. I still wonder how to implement a "final solution" on those blocks, so that they are not even downloaded perhaps along with its embedded images? Or would it break the layout too much?
))

**Comment by gildas-lormeau** [comment]
I'm glad this fixed the issue.

Regarding the userscript. Your script is executed before the page is displayed so it ensures the elements will never be visible. You could add the following script which would be run just after the page is loaded and would allow to remove the hidden elements. It should prevent images to be downloaded since `@run-at document-end` allows running the script before resources are downloaded according to the documentation.
`
// ==UserScript==
// ...
// @run-at document-end
// ==/UserScript==
document.querySelectorAll("#sidebar, .topbar").forEach(element => element.remove());`
`

I don't think it would break the layout. Actually, it's rare to break the layout because of this but I cannot take this risk with SingleFile.
